astropy/io/misc/tests/test_qdp.py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+import pytest
+from astropy.io.misc.qdp import read_table_qdp, write_table_qdp
+from astropy.table import Table, Column
+
+
+def test_get_tables_from_qdp_file():
+    import tempfile
+
+    example_qdp = """
+    ! Swift/XRT hardness ratio of trigger: XXXX, name: BUBU X-2
+    ! Columns are as labelled
+    READ TERR 1
+    READ SERR 2
+    ! WT -- hard data
+    !MJD            Err (pos)       Err(neg)        Rate            Error
+    53000.123456 2.37847222222222e-05    -2.37847222222222e-05   -0.212439       0.212439
+    55045.099887 1.14467592592593e-05    -1.14467592592593e-05   0.000000        0.000000
+    NO NO NO NO NO
+    ! WT -- soft data
+    !MJD            Err (pos)       Err(neg)        Rate            Error
+    53000.123456 2.37847222222222e-05    -2.37847222222222e-05   0.726155        0.583890
+    55045.099887 1.14467592592593e-05    -1.14467592592593e-05   2.410935        1.393592
+    NO NO NO NO NO
+    ! WT -- hardness ratio
+    !MJD            Err (pos)       Err(neg)        Rate            Error
+    53000.123456 2.37847222222222e-05    -2.37847222222222e-05   -0.292553       -0.374935
+    55045.099887 1.14467592592593e-05    -1.14467592592593e-05   0.000000        -nan
+    """
+
+    fd, path = tempfile.mkstemp()
```

Comment:
you can use the tmpdir or tmp_path fixture from pytest: https://docs.pytest.org/en/stable/tmpdir.html

Reply:
In this case, I would prefer using what the Python standard library gives me instead of a dependency that might change its behavior in the future, unless there is a compelling reason. After all, I just want a temporary file, and anything pytest can give me -as far as I understand- complicates the code instead of simplifying it